core/src/main/java/org/jenkins/ui/icon/IconSet.java

```diff
@@ -57,6 +66,43 @@ public static void initPageVariables(JellyContext context) {
         context.setVariable("icons", icons);
     }
 
+    private String prependTitleIfRequired(String icon, String title) {
+        if (StringUtils.isNotBlank(title)) {
+            return "<span class=\"jenkins-visually-hidden\">" + title + "</span>" + icon;
+        }
+        return icon;
+    }
+
+    public String getIonicon(String name, String title) {
```

Comment:
Are plugins likely going to want to add their own ionicon's? if so there will need to be another method which takes a `Class` or `Classloader` as an argument so as to locate the correct resource I believe.

(This can probably be a follow up)